githooks/post-checkout

```diff
@@ -2,5 +2,6 @@
 set -e
 
 if [ "$1" != "$2" ]; then # previous ref != new ref.
+  git clean -fd
```

Comment:
Hmm. So you want `git clean -fdx` if you want to clean `gitignore`d files, which I assume is the intention given the commit message? Otherwise the commit message should be updated.

Also, can we restrict the clean to just the `pkg` directory? Doing otherwise and wiping e.g. `build/builder_home` seems destructive.

Reply:
> can we restrict the clean to just the pkg directory? 

Done.

> So you want git clean -fdx if you want to clean gitignored files, which I assume is the intention given the commit message? Otherwise the commit message should be updated.

Omitting -x was intentional, though the commit message was faulty (fixed); I only care about "untracked files". These come about when your .gitignore directives change across branches so when switching to a branch where file X was not known about (and/or explicitly .gitignored), it shows up as untracked. If both branches ignore file X, we don't have to clean anything. In fact, nuking all .gitignored files will slow down `make` builds in the short term.